ivy/functional/extensions/layers.py

```diff
@@ -694,3 +694,72 @@ def sum_2N_1_n(n):
     ]
 
     return ivy.array(dn_low + dn_mid, dtype=dtype, out=out)
+
+
+@to_native_arrays_and_back
+@handle_out_argument
+@handle_nestable
+@handle_exceptions
+def hamming_window(
+    window_length: int,
+    /,
+    *,
+    periodic: Optional[bool] = True,
+    alpha: Optional[float] = 0.54,
+    beta: Optional[float] = 0.46,
+    dtype: Optional[Union[ivy.Array, ivy.NativeArray]] = None,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """Computes the Hamming window with window length window_length
+
+    Parameters
+    ----------
+    window_length
+        an int defining the length of the window.
+    periodic
+         If True, returns a window to be used as periodic function.
+         If False, return a symmetric window.
+    alpha
+        The coefficient alpha in the hamming window equation 
+    beta
+        The coefficient beta in the hamming window equation 
+    dtype
+        data type of the returned array.
+    out
+        optional output array, for writing the result to.
+
+    Returns
+    -------
+    ret
+        The array containing the window.
+
+    Examples
+    --------
+    >>> ivy.hamming_window(5)
+    ivy.array([0.0800, 0.3979, 0.9121, 0.9121, 0.3979])
+    >>> ivy.hamming_window(5, periodic=False)
+    ivy.array([0.0800, 0.5400, 1.0000, 0.5400, 0.0800])
+    >>> ivy.hamming_window(5, periodic=False, alpha=0.2, beta=2)
+    ivy.array([-1.8000,  0.2000,  2.2000,  0.2000, -1.8000])
+    """
+    if periodic is True:
+        window_length = window_length + 1
+        if window_length == 1:
```

Comment:
I guess this conditional clause should be before `window_length = window_length + 1` otherwise we are really handling input case of window_length = 0. 🙂

Also we need to add `if window_length == 0: return ivy.array([])` before everything.

lmk if unclear...

Reply:
No definitely clear, i guess i'm just tired enough to get confused over this 😅, hopefully this time's good